kuka_rsi_hw_interface/include/kuka_rsi_hw_interface/robot_control_node.hpp

```diff
@@ -17,38 +17,57 @@
 
 #include <string>
 #include <memory>
+#include <vector>
 
 #include "rclcpp/rclcpp.hpp"
 #include "kuka_rsi_hw_interface/kuka_hardware_interface.hpp"
 #include "kroshu_ros2_core/ROS2BaseLCNode.hpp"
 #include "sensor_msgs/msg/joint_state.hpp"
 
-
 namespace kuka_rsi_hw_interface
 {
+using CallbackReturn = rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn;
 
 class RobotControlNode : public kroshu_ros2_core::ROS2BaseLCNode
 {
 public:
   RobotControlNode(const std::string & node_name, const rclcpp::NodeOptions & options);
 
 private:
-  rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
-  on_configure(const rclcpp_lifecycle::State &) override;
-  rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
-  on_cleanup(const rclcpp_lifecycle::State &) override;
-  rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
-  on_activate(const rclcpp_lifecycle::State &) override;
-  rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
-  on_deactivate(const rclcpp_lifecycle::State &) override;
+  CallbackReturn on_configure(const rclcpp_lifecycle::State &) override;
+  CallbackReturn on_cleanup(const rclcpp_lifecycle::State &) override;
+  CallbackReturn on_activate(const rclcpp_lifecycle::State &) override;
+  CallbackReturn on_deactivate(const rclcpp_lifecycle::State &) override;
 
   std::unique_ptr<KukaHardwareInterface> kuka_rsi_hw_interface_;
 
-  void commandReceivedCallback(sensor_msgs::msg::JointState::ConstSharedPtr msg);
-  void Controlloop();
+  void commandReceivedCallback(sensor_msgs::msg::JointState::SharedPtr msg);
+  bool onRSIIPAddressChange(const std::string & rsi_ip_address);
+  bool onRSIPortAddressChange(int rsi_port);
+  bool onNDOFChange(int n_dof);
+
+  void ControlLoop();
 
   std::thread control_thread_;
-};
 
+  std::string rsi_ip_address_ = "";
+  int rsi_port_ = 0;
+  unsigned int n_dof_ = 6;
+
+  std::vector<double> initial_joint_pos_ = std::vector<double>(6, 0.0);
```

Comment:
consider moving this to the HWinterface, calculate difference and degree conversion there